### PR TITLE
introduce a flag to disable aten::cat in TE

### PR DESCRIPTION
**Pull request by Krovatkin**
introduce a flag to disable aten::cat in TE

### CODE REVIEW

File: torch/csrc/jit/passes/tensorexpr_fuser.cpp [hunk 1]
```diff
@@ -202,6 +206,11 @@ bool isSupported(Node* node) {
       }
     }
 
+    if (FLAGS_torch_jit_disable_cat &&
+        node->matches("aten::cat(Tensor[] tensors, int dim=0) -> Tensor")) {
```
Comment (eellison) [inline torch/csrc/jit/passes/tensorexpr_fuser.cpp:210]: Nit: all of the other uses of `cat` in this file use `aten::cat` as a symbol, and this uses aten::cat as a schema.

**Review by eellison — APPROVED** [review]
The lowering is just inefficient on CPU; IMO it would be better for the flag to only disable cat for cpu.

**Comment by walterddr** [comment]
seems like this PR broke clang-tidy @Krovatkin.

**Comment by heitorschueroff** [comment]
I'm unlanding this PR as it broke clang-tidy.

**Comment by eellison** [comment]
clang-tidy doesn't run on master, I don't think it's common practice to revert for a clang-tidy failure ? cc @mruberry

**Comment by mruberry** [comment]
> clang-tidy doesn't run on master, I don't think it's common practice to revert for a clang-tidy failure ? cc @mruberry

Let's follow-up with treehuggers.

**Pull request closed without merging**